src/Transformer/AbstractEnumTransformer.php

```diff
@@ -0,0 +1,48 @@
+<?php
+declare(strict_types=1);
+
+namespace Diaclone\Transformer;
+
+use Diaclone\Resource\ResourceInterface;
+
+abstract class AbstractEnumTransformer extends AbstractTransformer
+{
+    public function getPropertyValue($data, $property)
+    {
+        if (empty($property)) {
+            return $data;
+        }
+
+        if (is_array($data)) {
+            return $data[$property] ?? null;
+        }
+
+        // convert property to camelCase
+        $getter = 'get' . str_replace(' ', '', ucwords(str_replace(['-', '_'], ' ', $property)));
+        if (method_exists($data, $getter)) {
+            $enum = $data->$getter();
+            return $enum->getValue();
+        }
+
+        return $data->$property ?? null;
+    }
+    
+    public function transform(ResourceInterface $resource)
+    {
+        return $this->getPropertyValueFromResource($resource);
+    }
+
+    public function untransform(ResourceInterface $resource)
+    {
+        if ($resource->getData() === null) {
+            return null;
+        }
+
+        $data = $resource->getData();
+        $propertyTransform = $this->getObjectClass();
+
+        return $propertyTransform::byValue($data);
```

Comment:
@alexcoffee 
> This means I need to pass the bad value along instead of throwing exception.

How are you wanting it passed? As a string? Or an empty enum (a bit of a hack but possible)